qiskit_cold_atom/fermions/ffsim_backend.py

```diff
@@ -0,0 +1,388 @@
+            )
+        elif isinstance(op, Phase):
+            orbs = [qubit_indices[q] for q in qubits]
+            spatial_orbs = _get_spatial_orbitals(orbs, norb)
+            vec = _simulate_phase(
+                vec, np.array(op.params), spatial_orbs, norb=norb, nelec=nelec, copy=False
+            )
+        elif isinstance(op, FermionicGate):
+            orbs = [qubit_indices[q] for q in qubits]
+            spatial_orbs = _get_spatial_orbitals(orbs, norb)
+            ferm_op = _fermionic_op_to_fermion_operator(op.generator, spatial_orbs)
+            linop = ffsim.linear_operator(ferm_op, norb, nelec)
+            # TODO use ferm_op.values once it's available
+            scale = sum(abs(ferm_op[k]) for k in ferm_op)
+            vec = expm_multiply(-1j * linop, vec, traceA=scale)
+
+    result = {"statevector": vec}
+
+    if shots is None:
+        result["memory"] = []
+        result["counts"] = {}
+    else:
+        rng = np.random.default_rng(seed)
+        probs = np.abs(vec) ** 2
+        samples = rng.choice(np.arange(len(vec)), size=shots, replace=True, p=probs)
+        bitstrings = ffsim.indices_to_strings(samples, norb, nelec)
+        result["memory"] = bitstrings
+        result["counts"] = Counter(bitstrings)
+
+    return result
+
+
+def _get_initial_occupations(circuit: QuantumCircuit):
+    norb = circuit.num_qubits // 2
+    occ_a, occ_b = set(), set()
+    occupations = [occ_a, occ_b]
+    active_qubits = set()
+    for instruction in circuit.data:
+        if isinstance(instruction.operation, LoadFermions):
+            for q in instruction.qubits:
+                if q in active_qubits:
+                    raise ValueError(
+                        f"Encountered Load instruction on qubit {q} after it has "
+                        "already been operated on."
+                    )
+                spin, orb = divmod(circuit.qubits.index(q), norb)
+                # reverse index due to qiskit convention
+                occupations[spin].add(norb - 1 - orb)
+        else:
+            active_qubits |= set(instruction.qubits)
+    return tuple(occ_a), tuple(occ_b)
+
+
+def _get_spatial_orbitals(orbs: list[int], norb: int) -> list[int]:
+    assert len(orbs) % 2 == 0
+    alpha_orbs = orbs[: len(orbs) // 2]
+    beta_orbs = [orb - norb for orb in orbs[len(orbs) // 2 :]]
+    assert alpha_orbs == beta_orbs
+    # reverse orbitals due to qiskit convention
+    alpha_orbs = [norb - 1 - orb for orb in alpha_orbs]
+    return alpha_orbs
+
+
+def _simulate_hop(
+    vec: np.ndarray,
+    coeffs: np.ndarray,
+    target_orbs: list[int],
+    norb: int,
+    nelec: tuple[int, int],
+    copy: bool,
+) -> np.ndarray:
+    mat = np.zeros((norb, norb))
+    for i, val in zip(range(len(target_orbs) - 1), coeffs):
+        j, k = target_orbs[i], target_orbs[i + 1]
+        # TODO the following code makes the behavior match with FermionSimulator
```

Comment:
Is this still an issue ? Then it would be nice if we could fix this.

Reply:
I discovered that this is actually a bug in the existing FermionSimulator. See https://github.com/qiskit-community/qiskit-cold-atom/issues/92, which is fixed by https://github.com/qiskit-community/qiskit-cold-atom/pull/93.